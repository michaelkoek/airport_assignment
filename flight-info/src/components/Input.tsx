import React from "react";
import styled from "styled-components";

interface InputProps {
    name: string;
    type?: "text" | "number" | "search" | "password";
    disabled?: boolean;
    label?: string;
    placeholder?: string;
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => any;
}

const InputContainer = styled.section`
    padding: 10px 5px;
    width: 100%;
`;

const InputElement = styled.input`
    width: 100%;
    border-radius: 5px;
    padding: 10px;
    border: 2px solid var(--grey-storm);
`;

const Label = styled.label`
    margin-bottom: 5px;
    display: block;
    font-weight: 700;
`;

export const Input = ({
    name,
    type = "text",
    disabled = false,
    onChange,
    label,
    placeholder,
}: InputProps) => {
    return (
        <InputContainer>
            <Label htmlFor={name}>{label}</Label>
            <InputElement
                type={type}
                id={name}
                name={name}
                disabled={disabled}
                onChange={onChange}
                placeholder={placeholder}
            />
        </InputContainer>
    );
};
