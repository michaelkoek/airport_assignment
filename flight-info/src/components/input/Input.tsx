import React from "react";
import { InputContainer, InputElement, Label } from "./Input.styles";

interface InputProps {
    name: string;
    type?: "text" | "number" | "search" | "password";
    disabled?: boolean;
    label?: string;
    placeholder?: string;
    onChange: (e: React.ChangeEvent<HTMLInputElement>) => any;
}

export const Input = ({
    name,
    type = "text",
    disabled = false,
    onChange,
    label,
    placeholder,
}: InputProps) => {
    return (
        <InputContainer>
            <Label htmlFor={name}>{label}</Label>
            <InputElement
                type={type}
                id={name}
                name={name}
                disabled={disabled}
                onChange={onChange}
                placeholder={placeholder}
            />
        </InputContainer>
    );
};
