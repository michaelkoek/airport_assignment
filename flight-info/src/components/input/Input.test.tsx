import React from "react";
import { render } from "@testing-library/react";

import { Input } from "./Input";

describe("Input", () => {
    it("renders correctly as text input", () => {
        const { container } = render(
            <Input
                name="test"
                type="text"
                onChange={jest.fn()}
                placeholder="test"
            />
        );
        expect(container).toMatchSnapshot();
        expect(container).toBeVisible();
    });

    it("renders correctly with properties provided", () => {
        const { getByTestId } = render(
            <Input
                label="test label"
                name="text-input"
                type="text"
                disabled={true}
                onChange={jest.fn()}
                placeholder="text"
            />
        );
        const labelElement = getByTestId("labelname");
        const inputElement = getByTestId("input");

        expect(labelElement).toHaveTextContent("test label");
        expect(inputElement).toContainHTML("input");
        expect(inputElement).toHaveProperty("type");
        expect(inputElement).toHaveProperty("placeholder");
        expect(inputElement).toHaveProperty("value");
        expect(inputElement).toBeDisabled();
    });
});
