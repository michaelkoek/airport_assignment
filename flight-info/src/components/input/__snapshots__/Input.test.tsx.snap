// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly as text input 1`] = `
<div>
  <section
    class="sc-bcXHqe VcqUi"
  >
    <label
      class="sc-dkrFOg cIWZJf"
      data-testid="labelname"
      for="test"
    />
    <input
      class="sc-gswNZR ipnriv"
      data-testid="input"
      id="test"
      name="test"
      placeholder="test"
      type="text"
    />
  </section>
</div>
`;
